src/libraries/System.IO.FileSystem/tests/RandomAccess/FlushToDisk.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipes;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class RandomAccess_FlushToDisk : RandomAccess_Base<long>
+    {
+        // Setting this to false disables the ThrowsArgumentOutOfRangeExceptionForNegativeFileOffset() test
+        // which is not applicable to FlushToDisk() since FlushToDisk() does not take a file offset parameter.
+        protected override bool UsesOffsets => false;
+
+        // Setting this to false disables the ThrowsNotSupportedExceptionForUnseekableFile() test which is not
+        // applicable to FlushToDisk() since FlushToDisk() DOES in fact support unseekable files.
+        protected override bool ThrowsForUnseekableFile => false;
+
+        protected override long MethodUnderTest(SafeFileHandle handle, byte[] bytes, long fileOffset)
+        {
+            // NOTE: tests for checking how FlushToDisk() deals with invalid arguments (e.g. a null handle)
+            // are implemented in the base class and work by calling this "MethodUnderTest" which we override
+            // here to call the FlushToDisk() method we want to test.
+            RandomAccess.FlushToDisk(handle);
+            return 0;
+        }
+
+        [Fact]
+        public void UpdatesFileLastWriteTime()
+        {
+            // Save test file path so we can refer to the same file throughout the test.
+            string testFilePath = GetTestFilePath();
+
+            // Generate random bytes to write to file. To ensure that flushing works correctly for a variety of
+            // sizes, we want to test with buffers that are smaller than a page (e.g. just 1 byte) and buffers
+            // that are several times larger than a page (e.g. up to 10 pages).
```

Comment:
agreed, and you will probably need to use `[MemberData]` since you need to evaluate `Environment.SystemPageSize` to generate the list of variations for xunit.

example: https://github.com/dotnet/runtime/blob/d512f19ef3a506c01bb3a8588de48cb377472639/src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Match.Tests.cs#L1036